File: src/main/java/textGame/NpcFactory.java
package textGame;

import java.util.Random;

public class NpcFactory extends GameObjectCsvFactory {
    @Override
    public GameObject create(String[] npcInformation) {
        Npc npc = new Npc();

        npc.setName(npcInformation[0]);
        npc.setDescription(npcInformation[1]);
        npc.setDamage(Integer.parseInt(npcInformation[2]));

        npc.addRandomLoot();

        return npc;
    }

    @Override
    public String getCsvPath() {
        return "src/main/resources/npcs.csv";
    }
}

File: src/main/java/textGame/NpcFactory.kt
package textGame

class NpcFactory : GameObjectCsvFactory() {
    override fun create(npcInformation: Array<String>): GameObject {
        val npc = Npc()

        npc.setName(npcInformation[0])
        npc.setDescription(npcInformation[1])
        npc.damage = npcInformation[2].toInt()

        npc.addRandomLoot()

        return npc
    }

    override fun getCsvPath(): String {
        return "src/main/resources/npcs.csv"
    }
}